### PR TITLE
Changes in create and destroy credit functions to accept nr parameter

### PR DESCRIPTION
**Pull request by ydb242**
Signed-off-by: Yash Bhamare <yash.bhamare@seagate.com>

### CODE REVIEW

File: btree/btree.h [hunk 1]
```diff
@@ -158,13 +158,15 @@ enum m0_btree_opflag {
  * credit to @accum.
  */
 M0_INTERNAL void m0_btree_create_credit(const struct m0_btree_type *bt,
-					struct m0_be_tx_credit *accum);
+					struct m0_be_tx_credit *accum,
+					m0_bcount_t nr);
```
Comment (shashank-parulekar) [inline btree/btree.h:162]: Update the comment above to include description for nr.

File: btree/btree.h [hunk 2]
```diff
@@ -158,13 +158,15 @@ enum m0_btree_opflag {
...
 
 /**
  * Calculates the credit needed to destroy tree and adds this credit to @accum.
  */
 M0_INTERNAL void m0_btree_destroy_credit(struct m0_btree *tree,
-					 struct m0_be_tx_credit *accum);
+					 struct m0_be_tx_credit *accum,
+					 m0_bcount_t nr);
```
Comment (shashank-parulekar) [inline btree/btree.h:169]: Update the comment above to include description for nr.

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

**Review by NikhilBirgade — APPROVED** [review]

File: btree/btree.h (hunk 1 above)
Comment (ydb242) [reply btree/btree.h:162]: Done

File: btree/btree.h (hunk 2 above)
Comment (ydb242) [reply btree/btree.h:170]: Done

File: btree/btree.c [hunk 3]
```diff
@@ -5327,19 +5327,24 @@ M0_INTERNAL void m0_btree_update_credit2(const struct m0_btree_type *type,
...
 
 M0_INTERNAL void m0_btree_create_credit(const struct m0_btree_type *bt,
-					struct m0_be_tx_credit *accum)
+					struct m0_be_tx_credit *accum,
+					m0_bcount_t nr)
 {
 	const struct node_type *nt = btree_nt_from_bt(bt);
 	int size = nt->nt_create_delete_credit_size();
-	m0_be_tx_credit_add(accum, &M0_BE_TX_CREDIT(1, size));
+	struct m0_be_tx_credit cred = M0_BE_TX_CREDIT(1, size);
```
Comment (RadhaGulhane13) [inline btree/btree.c:5335]: alignment

Comment (ydb242) [reply btree/btree.c]: Done

**Review by RadhaGulhane13 — APPROVED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**